### PR TITLE
Ctrl state airspeed

### PR DESCRIPTION
**Pull request by tumbili**

### CODE REVIEW

**Comment by tumbili** [comment]
@LorenzMeier The relevant fix is in the first commit. The second commit is just code style, not sure if I should fix code style for every file that I modify.

**Comment by tumbili** [comment]
@yyassa Review would be great!

File: src/modules/attitude_estimator_q/attitude_estimator_q_main.cpp [hunk 1]
```diff
@@ -623,7 +623,13 @@ void AttitudeEstimatorQ::task_main()
 		ctrl_state.yaw_rate = _lp_yaw_rate.apply(_rates(2));
 
 		/* Airspeed - take airspeed measurement directly here as no wind is estimated */
-		ctrl_state.airspeed = _airspeed.indicated_airspeed_m_s;
+		if (PX4_ISFINITE(_airspeed.indicated_airspeed_m_s) && hrt_absolute_time() - _airspeed.timestamp < 1e6) {
```
Comment (LorenzMeier) [inline src/modules/attitude_estimator_q/attitude_estimator_q_main.cpp]: Can you add a "_airspeed.timestamp > 0 &&" here? Else it takes a second on boot where the airspeed is first valid.

**Comment by LorenzMeier** [comment]
The code style fix is fine. But we need to add it to Tools/check_code_style.sh as well (it was not run by default for that module yet).

File: src/modules/ekf_att_pos_estimator/ekf_att_pos_estimator_main.cpp [hunk 2]
```diff
@@ -896,7 +903,14 @@ void AttitudePositionEstimatorEKF::publishControlState()
 	//_ctrl_state.airspeed = sqrt(pow(_ekf->states[4] -  _ekf->states[14], 2) + pow(_ekf->states[5] - _ekf->states[15], 2) + pow(_ekf->states[6], 2));
 	// the line above was introduced by the control state PR. The airspeed it gives is totally wrong and leads to horrible flight performance in SITL
 	// and in outdoor tests
-	_ctrl_state.airspeed = _airspeed.true_airspeed_m_s;
+	if (PX4_ISFINITE(_airspeed.indicated_airspeed_m_s) && hrt_absolute_time() - _airspeed.timestamp < 1e6) {
```
Comment (LorenzMeier) [inline src/modules/ekf_att_pos_estimator/ekf_att_pos_estimator_main.cpp]: Also add the "_airspeed.timestamp > 0 &&" check here.

**Comment by LorenzMeier** [comment]
This brings us back to what we had earlier, but doesn't cover the issue of broken / blocked sensors.

**Comment by yyassa** [comment]
Are you still getting issues with the "estimated" version of airspeed or is that resolved? 
I would definitely go for the more robust version..it can handle sensor failure and also gives less time delay than the low-pass filtered direct sensor input..

**Comment by LorenzMeier** [comment]
@yyassa The "robust" version didn't work at all, so there is some disconnect here. Are you sure you guys are really flying something without a significant diff to our setup?

**Comment by yyassa** [comment]
@LorenzMeier Yes, I am sure it is identical, also it does not make sense to give completely wrong behaviour as long as local velocities and wind estimates are relatively sane.. 
If you tell me what problem exactly you are facing maybe I can help debug. 

(I thought that the issue only arised when the wrong modules were running)

**Comment by LorenzMeier** [comment]
I'll handle this myself.

**Comment by LorenzMeier** [comment]
My comments were outdated, the > 0 is already in there. Great!

**Comment by LorenzMeier** [comment]
Rebased and applied.

**Pull request closed without merging**